acceptance/features/migration.feature

```diff
@@ -0,0 +1,25 @@
+Feature: operator can migrate/adopt Redpanda helm base deployment
+
+  @skip:gke @skip:aks @skip:eks
+  Scenario: Adopt or migrate from stand alone helm chart release to Redpanda custom resource
+        Given there is "test-migrate-from-helm" Redpanda helm release with values:
+        """
+        fullnameOverride: name-override
+        """
+
+        When I apply Kubernetes manifest:
+        """
+    # tag::redpanda-custom-resource-manifest[]
+        # This manifest is a copy of Redpanda release helm values
+        ---
+        apiVersion: cluster.redpanda.com/v1alpha2
+        kind: Redpanda
+        metadata:
+          name: test-migrate-from-helm
+        spec:
+          clusterSpec:
+            fullnameOverride: name-override
+    # end::redpanda-custom-resource-manifest[]
+        """
+
+        Then "test-migrate-from-helm" Redpanda custom resource is in ready state
```

Comment:
I think we'll want a slightly more strict assertion with this. My immediate thoughts are:
- What happens if I run `helm uninstall`?
- Are there now two redpanda clusters or just one?


Thinking about it a bit, we might want to use a static version of the helm chart in the v2.4.x and v.2.3.x branches. That way we could assert that the Statefulset exists, hasn't had it's Generation changed and has an OwnerRefence to the Redpanda CR.

Reply:
> What happens if I run helm uninstall?

It depends on the type of Storage Class used for the deployment. Resources would be deleted, but quickly re-created by operator. Certificates if not referenced by SecretRef would be re-created too.

Reply:
> Are there now two redpanda clusters or just one?

One as operator will adopt all resourced.

Reply:
> Thinking about it a bit, we might want to use a static version of the helm chart in the v2.4.x and v.2.3.x branches. That way we could assert that the Statefulset exists, hasn't had it's Generation changed and has an OwnerRefence to the Redpanda CR.

As this PR is opened against main I will do everything except static Redpanda helm chart version.

Reply:
How in the acceptance test suite I could store fact and then assert it later. Mainly the Generation number before Redpanda custom resource creation and after.